__tests__/inputValidation.test.js

```diff
@@ -0,0 +1,71 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '..';
+import mockUsers from './mockData/mockUsers';
+
+chai.use(chaiHttp);
+
+const { expect } = chai;
+const url = '/api/v1';
+describe('User tests', () => {
+  it('should return status 400 when all required fields are missing', (done) => {
+    chai.request(app)
+      .post(`${url}/auth/signup`)
+      .send(mockUsers[0])
+      .end((err, res) => {
+        expect(res.status).eql(400);
+        expect(res.body).to.have.property('error');
+        expect(res.body.error.email).eql('Email is required');
+        expect(res.body.error.firstName).eql('First name is required');
+        expect(res.body.error.lastName).eql('Last name is required');
+        expect(res.body.error.password).eql('Password is required');
+        done();
+      });
+  });
+  it('should return status 400 if the email input is invalid', (done) => {
+    chai.request(app)
+      .post(`${url}/auth/signup`)
+      .send(mockUsers[1])
+      .end((err, res) => {
+        expect(res.status).eql(400);
+        expect(res.body).to.have.property('error');
+        expect(res.body.error.email).eql('Please input a valid email address');
+        done();
+      });
+  });
+  it('should return status 400 if the password input does not match required expectation', (done) => {
+    chai.request(app)
+      .post(`${url}/auth/signup`)
+      .send(mockUsers[2])
+      .end((err, res) => {
+        expect(res.status).eql(400);
+        expect(res.body).to.have.property('error');
+        expect(res.body.error.password)
+          .eql('Password must contain at least one uppercase letter, one lowercase letter and one numeric digit');
+        done();
+      });
+  });
+  it('should return status 400 if the password length is less than 8 characters', (done) => {
+    chai.request(app)
+      .post(`${url}/auth/signup`)
+      .send(mockUsers[3])
+      .end((err, res) => {
+        expect(res.status).eql(400);
+        expect(res.body).to.have.property('error');
+        expect(res.body.error.password).eql('Password must be at least 8 characters');
+        done();
+      });
+  });
+  it('should successfully signup a user if the parameters are valid', (done) => {
+    chai.request(app)
+      .post(`${url}/auth/signup`)
+      .send(mockUsers[4])
+      .end((err, res) => {
+        expect(res.status).eql(201);
+        expect(res.body).to.have.property('user');
+        expect(res.body.user).to.have.property('firstName');
+        expect(res.body.user).to.have.property('lastName');
```

Comment:
This is still not fixed. 

Like I said, it is not enough to check that a response has a property, you need to be sure that the value of that property is what you expect it to be. So check that the `mockUsers[4]`'s firlstname, lastname, etc match what was returned in the response.

Something like:
`expect(res.body.user.firstName).eql(mockUsers[4].firstName);`

Does this make sense now?

Reply:
Yes. It does, I’ll fix that now